File: Prog/Min_Moves/Minimum_Moves.py
def minMoves2(nums):
    nums.sort()  # Sort the array
    median = nums[len(nums) // 2]  # Find the median element
    
    moves = 0
    for num in nums:
        moves += abs(num - median)  # Calculate the absolute difference to the median
    
    return moves

# Example usage
nums = [1, 2, 3]
result = minMoves2(nums)
print(result)

File: Prog/Min_Moves/Minimum_Moves.cpp
#include <iostream>
#include <algorithm>
#include <vector>
#include <cmath>
using namespace std;

int minMoves2(vector<int>& nums) {
    sort(nums.begin(), nums.end());
    int median = nums[nums.size() / 2];
    
    int moves = 0;
    for (int i = 0; i < nums.size(); i++) {
        moves += abs(nums[i] - median);
    }
    
    return moves;
}

int main() {
    vector<int> nums = {1, 2, 3};
    int result = minMoves2(nums);
    cout << result << endl;
    
    return 0;
}

